OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -154,16 +155,26 @@ void receivedInAppMessageJson(@NonNull JSONArray json) throws JSONException {
         // Cache copy for quick cold starts
         OneSignalPrefs.saveString(OneSignalPrefs.PREFS_ONESIGNAL,
            OneSignalPrefs.PREFS_OS_CACHED_IAMS, json.toString());
-        processInAppMessageJson(json);
+        ArrayList<OSInAppMessage> newMessages = processInAppMessageJson(json);
+
+        List<OSInAppMessage> updatedRedisplayMessages = inAppMessageRepository.updateInAppMessage(newMessages, redisplayInAppMessages);
```

Comment:
The source of `newMessages` is driven directly by the results from the  `on_session` network call . However the results from `on_session` is drive by the backend and if the player fits into the Segments defined there. So this means the same IAM could fall in and out of Segment multiple times which would result in the on_session not always have it either.

So we don't reset the display stats noted above we should only delete these local records after a long period of time to be safe or not at all. Can we set this to 6 months to be safe?

Reply:
oh ok, i will address this on here and iOS. 6 Months seems like a logic time period